reedstrm/baking_workflow_and_errors.md

```diff
@@ -0,0 +1,120 @@
+`pending`, and process them on a delayed schedule. An interface for directly triggering baking for a given book in the pending state will be provided, but a time-based process will also "sweep up" books in this state. However, only the most recent minor version will be baked - when multiple pages and edited and published, each publish event will set the book with the new minor version to `pending`, while setting the previously pending minor version to `raw_only`.
+
+## Reporting Baking State
+
+The state of baking of a given book will be available in the `/extras` page for that book. This will be used to drive additional UI in the `More Information` tab, where the "Print Style" is currently reported. If the state is `pending` or `stale_recipe`, a link to trigger immediate baking will be provided (this is a `POST`). Also, the most recent history of baking, with error messages, will be available here.
+
+ In addition, a global "state of baking" page will give an overview of all books and their current baked state. This may it in with the publishing administrative pages, under `/a` Again, any that are not current will have links  available to attempt re-baking.
+
+|    result     |    state      |  color/shape    |  emoji             |
+|:-------------:|:-------------:|:---------------:|:------------------:|
+|  Done/Success | up_to_date    |  Green/circle   | :white_check_mark: / :heavy_check_mark: |
+|  Pending      | stale_recipe  |  Yellow/square  | :warning:          |
+|  Pending      | stale_content |  Yellow/square  | :warning:          |
+|  Fallback     | stale_recipe  |  Red/triangle   | :x:                |
+|  Fail         | stale_content |  Red/triangle   | :x:                |
+
+Pending conditions come about via injection of a new recipe (`stale_recipe`) or publishing of a new page version, causing book minor version rev (`stale_content`). Retransform of CNXML to HTML would also cause a `Pending/stale_content` state. These `pending` conditions are presumed to be resolved by an upcoming baking run.
+
+Red Conditions (`Fallback` and `Fail`) indicate that an error occurred while baking, and new bits (recipe or content) need to be uploaded to resolve the condition. The content available for students to see is not current.
+
+
+
+ ## Development Workflow  - Recipe Injection
+
+Testing of recipes during development of new recipes requires the ability to "inject" the recipe into the dev or testing server, so that the event-based asynchronous baking process will pick them up and `re-bake` the books of interest. In order to support the recipe selection described above, recipe devs need to be able to  replace the recipe associated with a given `print_style`, as well as provide a per-book custom recipe in place of a style-based one.
+
+We will create a cli tool similar to `cnx-archive-inject-resource` that is specialized for storing recipes in the archive.  This needs to handle 3 (and a half) cases:
```

Comment:
Could this be a web form (authenticated)? That way QA/CM's can use it as well without having to set up code and ssh access. It could be limited to non-production deployments.